stan/math/prim/mat/prob/categorical_logit_rng.hpp

```diff
@@ -0,0 +1,56 @@
+#ifndef STAN_MATH_PRIM_MAT_PROB_CATEGORICAL_LOGIT_RNG_HPP
+#define STAN_MATH_PRIM_MAT_PROB_CATEGORICAL_LOGIT_RNG_HPP
+
+#include <boost/random/uniform_01.hpp>
+#include <boost/random/variate_generator.hpp>
+#include <stan/math/prim/scal/err/check_finite.hpp>
+#include <stan/math/prim/mat/err/check_simplex.hpp>
+#include <stan/math/prim/mat/fun/cumulative_sum.hpp>
+#include <stan/math/prim/mat/fun/softmax.hpp>
+#include <stan/math/prim/mat/fun/Eigen.hpp>
+
+namespace stan {
+  namespace math {
+
+    /**
+     * Return a draw from a Categorical distribution given a
+     * a vector of unnormalized log probabilities and a psuedo-random
+     * number generator.
+     *
+     * This is a convenience wrapper around
+     * <code>categorical_rng(softmax(beta), rng)</code>.
+     *
+     * @tparam RNG Type of pseudo-random number generator.
+     * @param beta Vector of unnormalized log probabilities.
+     * @param rng Pseudo-random number generator.
+     * @return Categorical random variate
+     */
+    template <class RNG>
+    inline int
+    categorical_logit_rng(const Eigen::Matrix<double, Eigen::Dynamic, 1>& beta,
```

Comment:
If you're already getting in here, would you mind

* removing all the blank lines---it just makes the code hard to read
* lining up `RNG` under `const` in the arguments? 
* using `Eigen::VectorXd` instead of `Eigen::Matrix<double, Eigen::Dynamic, 1>`